Zend/zend_vm_def.h

```diff
@@ -9597,46 +9597,89 @@ ZEND_VM_C_LABEL(try_again):
 	}
 }
 
-ZEND_VM_HANDLER(204, ZEND_FRAMELESS_ICALL_0, UNUSED, UNUSED)
+ZEND_VM_HANDLER(204, ZEND_FRAMELESS_ICALL_0, UNUSED, UNUSED, SPEC(OBSERVER))
 {
 	USE_OPLINE
 	SAVE_OPLINE();
+
+#if !ZEND_VM_SPEC || ZEND_OBSERVER_ENABLED
+	if (ZEND_OBSERVER_ENABLED) {
+		zend_function *fbc = ZEND_FLF_FUNC(opline);
+		if (UNEXPECTED(zend_observer_handler_is_unobserved(ZEND_OBSERVER_DATA(fbc)) == false)) {
+			zend_execute_data *call = _zend_vm_stack_push_call_frame_ex(zend_vm_calc_used_stack(0, fbc), ZEND_CALL_NESTED_FUNCTION, fbc, 0, NULL);
```

Comment:
Can't we avoid call to ``zend_vm_calc_used_stack(0, fbc)``. The size of the stack should be known statically.
May be I'm wrong... I don't remember when we got ``zend_internal_function->T`` and what does it mean...

Reply:
It's the temporary the observers use to store the prev_observed_frame.
For FLF functions we can probably hardcode it to `(ZEND_CALL_FRAME_SLOT + X + 1) * sizeof(zval)`, at least for now given that we don't have a mechanism to dynamically allocate temps for arbitrary functions.

This is outside the hot path of not-observed functions, so I don't consider this as critical, but can change it.